EKrut-Common/src/ekrut/net/ShipmentRequest.java

```diff
@@ -8,27 +8,27 @@
 public class ShipmentRequest implements Serializable {
 
 	private static final long serialVersionUID = 5704922451119200259L;
-	private OrderStatus action;
+	private OrderStatus status;
+	private ShipmentRequestType action;
 	private int orderId;
 	private String clientAddress;
 	private LocalDateTime date;
+	private String area;
 	
-	public ShipmentRequest(int orderId, String clientAddress, LocalDateTime date) {
-		this.action = OrderStatus.AWAITING_DELIVERY;
-		this.orderId = orderId;
-		this.clientAddress = clientAddress;
-		this.date = date;
-	}
 	
+	// Worker Confirm deliveries.
+	public ShipmentRequest(ShipmentRequestType action, String area) {
+		this.action = action;
+		this.area = area;
+	}
 	
 	public ShipmentRequest(OrderStatus awaitingDelivery, int orderId) {
-		this.action = awaitingDelivery;
+		this.status = awaitingDelivery;
```

Comment:
You need to set the action as well. Any request must always specify what action it's requesting.